memCache/entryCache.go

```diff
@@ -433,3 +215,216 @@ func (e *Entry) timeToUpdate() bool {
 	}
 	return false
 }
+
+type EntryCache struct {
+	log            *log.Logger
+	clk            clock.Clock
+	requestTimeout time.Duration
+	entries        map[string]*Entry   // one-to-one map keyed on name -> entry
+	lookupMap      map[[32]byte]*Entry // many-to-one map keyed on sha256 hashed OCSP requests -> entry
```

Comment:
Can you explain more about why you need this secondary map? By my reading, it looks like LookupResponse gets the full OCSP request and could look up in the entries map based on that?